pkg/reconciler/instances/ory/action.go

```diff
@@ -113,19 +127,16 @@ func (a *preReconcileAction) Run(context *service.ActionContext) error {
 			return errors.Wrap(err, "failed to update db credentials data for Ory Hydra")
 		}
 
-		if len(newSecretData) == 0 {
+		if !isUpdate(newSecretData) {
 			logger.Info("Ory DB secret is the same as values, no need to update")
 		} else {
 			logger.Info("Ory DB secret is different than values, updating it")
 			dbSecretObject.StringData = newSecretData
+			rolloutHydra = true
```

Comment:
If we want to deal with it this way we should also do opposite and set it to `false` when it is not an update.